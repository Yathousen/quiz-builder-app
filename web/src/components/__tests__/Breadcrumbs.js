import React from 'react';
import renderer from 'react-test-renderer';
import { Breadcrumbs } from '../';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('Default Breadcrumbs', () => {
  expect(renderer.create(<Breadcrumbs />).toJSON()).toMatchSnapshot();
});

it('Extended Breadcrumbs', () => {
  expect(renderer.create(<Breadcrumbs data={[{ name: 'Extended' }]} />).toJSON()).toMatchSnapshot();
});
