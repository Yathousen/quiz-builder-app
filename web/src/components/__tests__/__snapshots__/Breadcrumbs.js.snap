// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Breadcrumbs 1`] = `
<div
  className="flex items-center py-4 overflow-y-auto whitespace-nowrap w-11/12 md:w-8/12 mx-auto md:text"
>
  <a
    className="text-gray-600 dark:text-gray-200 cursor-pointer"
    onClick={[Function]}
  >
    <svg
      className="w-5 h-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
      />
    </svg>
  </a>
</div>
`;

exports[`Extended Breadcrumbs 1`] = `
<div
  className="flex items-center py-4 overflow-y-auto whitespace-nowrap w-11/12 md:w-8/12 mx-auto md:text"
>
  <a
    className="text-gray-600 dark:text-gray-200 cursor-pointer"
    onClick={[Function]}
  >
    <svg
      className="w-5 h-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
      />
    </svg>
  </a>
  <span
    className="mx-5 text-gray-500 dark:text-gray-300"
  >
    <svg
      className="w-5 h-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
        fill-rule="evenodd"
      />
    </svg>
  </span>
  <a
    className="text-gray-600 dark:text-gray-200 cursor-pointer hover:underline"
    onClick={[Function]}
  >
    Extended
  </a>
</div>
`;
